src/land/smap9km_ssm2ioda.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python3
+#
+# (C) Copyright 2021 EMC/NCEP/NWS/NOAA
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+import time, os, sys
+import argparse
+import netCDF4 as nc
+import numpy as np
+import re
+from datetime import datetime, timedelta
+from pathlib import Path
+
+IODA_CONV_PATH = Path(__file__).parent/"@SCRIPT_LIB_PATH@"
+if not IODA_CONV_PATH.is_dir():
+    IODA_CONV_PATH = Path(__file__).parent/'..'/'lib-python'
+sys.path.append(str(IODA_CONV_PATH.resolve()))
+
+import ioda_conv_engines as iconv
+from collections import defaultdict, OrderedDict
+from orddicts import DefaultOrderedDict
+
+locationKeyList = [
+    ("latitude", "float"),
+    ("longitude", "float"),
+    ("dateTime", "string")
+]
+
+obsvars = {
+    'soil_moisture': 'soilMoistureVolumetric',
+}
+
+AttrData = {
+}
+
+DimDict = {
+}
+
+VarDims = {
+    'soilMoistureVolumetric': ['nlocs'],
+}
+
+
+class smap(object):
+    def __init__(self, filename, mask):
+        self.filename = filename
+        self.mask = mask
+        self.varDict = defaultdict(lambda: defaultdict(dict))
+        self.outdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+        self.varAttrs = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+        self._read()
+
+    # Open input file and read relevant info
+    def _read(self):
+        # set up variable names for IODA
+        for iodavar in ['soilMoistureVolumetric']:
+            self.varDict[iodavar]['valKey'] = iodavar, iconv.OvalName()
+            self.varDict[iodavar]['errKey'] = iodavar, iconv.OerrName()
+            self.varDict[iodavar]['qcKey'] = iodavar, iconv.OqcName()
+            self.varAttrs[iodavar, iconv.OvalName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OerrName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OqcName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OvalName()]['units'] = 'm3 m-3'
+            self.varAttrs[iodavar, iconv.OerrName()]['units'] = 'm3 m-3'
+            self.varAttrs[iodavar, iconv.OqcName()]['units'] = 'm3 m-3'
```

Comment:
Yes, when I used "unitless", ioda-validation.x shows error. So, if I put unit as "m3 m-3". ioda-validation.x does not have errors. Yes, I can change back to "unitless"  as my previous python converter. Just let me know I can ignore ioda-validation.x error and I will change back @srherbener

Reply:
For now it is okay to ignore the error when you use "unitless" on the PreQC variables.

Since we haven't switched over to the JEDI Conventions yet (in an upcoming code sprint) it is okay to ignore most of the errors/warnings from the validator. The usefulness for the validator now is to give you an idea of what will need to be changed when we do the code sprint.

Reply:
got it, thank you @srherbener. I will change it back.